server/app/app.py

```diff
@@ -1,26 +1,189 @@
 import os
 import datetime
 
-from flask import Flask
+from flask import Flask, redirect, current_app, make_response, render_template
+from flask import Blueprint, request, send_from_directory
 from flask_caching import Cache
 from flask_compress import Compress
 from flask_cors import CORS
+from flask_restful import Api, Resource
 
-from server.app.rest_api.rest import get_api_resources
-from server.app.util.utils import Float32JSONEncoder
-from server.app.web import webapp
+from http import HTTPStatus
+
+import server.common.rest as common_rest
+from server.common.utils import path_join, Float32JSONEncoder
+from server.data_common.matrix_loader import MatrixDataLoader, MatrixDataType, MatrixDataCacheManager
+
+from functools import wraps
+
+webbp = Blueprint("webapp", "server.common.web", template_folder="templates")
+
+
+@webbp.route("/")
+def index():
+    if current_app.data is None:
+        return dataroot_index()
+
+    dataset_title = current_app.app_config.title
+    scripts = current_app.app_config.scripts
+    return render_template("index.html", datasetTitle=dataset_title, SCRIPTS=scripts)
+
+
+@webbp.route("/favicon.png")
+def favicon():
+    return send_from_directory(os.path.join(webbp.root_path, "static/img/"), "favicon.png")
+
+
+def get_data_adaptor(dataset):
```

Comment:
dead code?

Reply:
Yes, it was dead code, but bringing it back in this patch for another purpose.